enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,221 @@
+
+[More info](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/multi-tenant-alerting.md)
+
+
+- Replace confimgaps with CRD:
+  
+```
+  type Config struct {
+	Images                               *Images `json:"-"`
+	RemoteWrite                          bool    `json:"-"`
+	CollectionProfilesFeatureGateEnabled bool    `json:"-"`
+
+	ClusterMonitoringConfiguration *ClusterMonitoringConfiguration `json:"-"`
+	UserWorkloadConfiguration      *UserWorkloadConfiguration      `json:"-"`
+}
+```
+
+We will strive to maintain the previous structure as much as possible while adapting it to OpenShift API conventions. 
+
+
+```
+type ClusterMonitoringConfiguration struct {
+	AlertmanagerMainConfig *AlertmanagerMainConfig `json:"alertmanagerMain,omitempty"`
+	UserWorkloadEnabled *bool `json:"enableUserWorkload,omitempty"`
+	HTTPConfig *HTTPConfig `json:"http,omitempty"`
+	K8sPrometheusAdapter *K8sPrometheusAdapter `json:"k8sPrometheusAdapter,omitempty"`
+	MetricsServerConfig *MetricsServerConfig `json:"metricsServer,omitempty"`
+	KubeStateMetricsConfig *KubeStateMetricsConfig `json:"kubeStateMetrics,omitempty"`
+	PrometheusK8sConfig *PrometheusK8sConfig `json:"prometheusK8s,omitempty"`
+	PrometheusOperatorConfig *PrometheusOperatorConfig `json:"prometheusOperator,omitempty"`
+	PrometheusOperatorAdmissionWebhookConfig *PrometheusOperatorAdmissionWebhookConfig `json:"prometheusOperatorAdmissionWebhook,omitempty"`
+	OpenShiftMetricsConfig *OpenShiftStateMetricsConfig `json:"openshiftStateMetrics,omitempty"`
+	TelemeterClientConfig *TelemeterClientConfig `json:"telemeterClient,omitempty"`
+	ThanosQuerierConfig *ThanosQuerierConfig `json:"thanosQuerier,omitempty"`
+	NodeExporterConfig NodeExporterConfig `json:"nodeExporter,omitempty"`
+	MonitoringPluginConfig *MonitoringPluginConfig `json:"monitoringPlugin,omitempty"`
+}
+```
+
+
+Each component within the ConfigMap will be migrated to the OpenShift API in separate PRs. This approach allows for a thorough review, improvement, and modification of each ConfigMap component to ensure it aligns with OpenShift API standards. As part of this process, types will be modified, outdated elements will be removed and names and configurations will be refined.
+
+
+### Migration path
+
+
+- Switch mecanishm
+- Both ConfigMap and CRD will coexist for a while until the transition is complete.
+- CRD and ConfigMap will be merged into a single structure, this structure will config CMO
+- Extensive testing will be required to validate the new behavior.
+- If there are different fields in CRD and ConfigMap, the field in the ConfigMap will take precedence.
+- Some fields in CRD and ConfigMap diverge due to API requirements. Helper functions will be needed to resolve this.
+- The end user will be able to choose to use ConfigMap or CRD at any time, keeping in mind that the fields in the CRD will gradually expand.
```

Comment:
Will this be a conscious choice? Is there some explicit switch for this or is it just a case of if one or the other exists?

Reply:
does user still can choose to use ConfigMap or CRD after the feature GA?

Reply:
The feature gate acts as the switch for this functionality, which is a deliberate choice since the CR will continue to expand until it is fully developed. After the feature reaches GA, users can choose between the CR and ConfigMap, unless there is a strong consensus to remove the ConfigMap